packages/lodestar/src/network/reqresp/encodingStrategies/sszSnappy/decode.ts

```diff
@@ -138,9 +144,18 @@ function deserializeSszBody<T extends RequestOrIncomingResponseBody>(
   try {
     if (options?.deserializeToTree) {
       const typeTree = (type as unknown) as CompositeType<Record<string, unknown>>;
-      return (typeTree.createTreeBackedFromBytes(bytes) as unknown) as T;
+      const treeBacked = typeTree.createTreeBackedFromBytes(bytes) as unknown;
+      if (options?.cacheBytes) {
+        return (new Proxy({treeBacked, bytes}, cachedTreeBackedProxyHandler) as unknown) as T;
```

Comment:
Needed another proxy as tree's proxy wasn't letting assign `bytes` without going through the tree setters. May be post ssz v2 merge, it could be removed.